src/v/kafka/client/broker.h

```diff
@@ -82,6 +82,13 @@ class broker : public ss::enable_lw_shared_from_this<broker> {
                 return ss::make_exception_future<Ret>(
                   broker_error(_node_id, error_code::broker_not_available));
             })
+          .handle_exception_type([this](const std::system_error& e) {
+              if (net::is_reconnect_error(e)) {
+                  return ss::make_exception_future<Ret>(
+                    broker_error(_node_id, error_code::broker_not_available));
+              }
+              return ss::current_exception_as_future<Ret>();
```

Comment:
Does this work as intended? I'm not entirely sure, I checked the seastar source says it returns what `std::current_exception` would, and I believe that only works when invoked within catch clauses. If it doesn't work i'm pretty sure `ss::make_exception_future<Ret>(e)` would do what you want.

Reply:
I think you're right, this isn't doing what I want it to do.